### PR TITLE
Custodian emits a new asset-receive-complete event to notification subscribers

### PR DESCRIPTION
**Pull request by ffranr**
Fixes https://github.com/lightninglabs/taproot-assets/issues/650

### CODE REVIEW

File: tapgarden/custodian.go [hunk 1]
```diff
@@ -698,6 +733,19 @@ func (c *Custodian) RegisterSubscriber(receiver *fn.EventReceiver[fn.Event],
...
+// publishSubscriberStatusEvent publishes an event to all status events
+// subscribers.
+func (c *Custodian) publishSubscriberStatusEvent(event fn.Event) {
+	// Lock the subscriber mutex to ensure that we don't modify the
+	// subscriber map while we're iterating over it.
+	c.statusEventsSubsMtx.Lock()
+	defer c.statusEventsSubsMtx.Unlock()
+
+	for _, sub := range c.statusEventsSubs {
+		sub.NewItemCreated.ChanIn() <- event
```
Comment (GeorgeTsagk) [inline tapgarden/custodian.go:745]: is this non-blocking?

if it's blocking then maybe worth considering bringing down the ctx to this function to ensure healthy exit

File: itest/send_test.go [hunk 2]
```diff
@@ -754,6 +754,10 @@ func testReattemptFailedReceiveUniCourier(t *harnessTest) {
 	// proof(s).
 	t.Logf("Attempting to confirm asset received by receiver node")
 	AssertNonInteractiveRecvComplete(t.t, receiveTapd, 1)
+
+	// Confirm that the sender tapd node eventually receives the asset
+	// transfer and publishes an asset recv complete event.
+	assertAssetRecvCompleteEvent(t, ctxb, 5*time.Second, eventNtfns)
```
Comment (GeorgeTsagk) [inline itest/send_test.go:760]: Maybe could go a tiny bit further and also check receiving address?

**Review by GeorgeTsagk — APPROVED** [review]
Looks good! couple of minor comments

File: tapgarden/custodian.go (hunk 1 above)
Comment (ffranr) [reply tapgarden/custodian.go:745]: Good point! I'll use `fn.SendOrQuit`.

File: itest/send_test.go (hunk 2 above)
Comment (ffranr) [reply itest/send_test.go]: Good idea! I've added the recv address as an arg to `assertAssetRecvCompleteEvent`.

File: tapgarden/custodian.go (hunk 1 above)
Comment (jharveyb) [reply tapgarden/custodian.go]: Looks like we'd want to use `SendOrQuit` for all event publishing, not just here? Could be follow-up PR.

File: tapgarden/custodian.go [hunk 3]
```diff
@@ -458,6 +488,16 @@ func (c *Custodian) inspectWalletTx(walletTx *lndclient.Transaction) error {
 				log.Errorf("unable to import proofs: %v", err)
 				return
 			}
+
+			// At this point the "receive" process is complete. We
+			// will now notify all status event subscribers.
+			recvCompleteEvent := NewAssetReceiveCompleteEvent(*addr, op)
```
Comment (jharveyb) [inline tapgarden/custodian.go:494]: nit: line length; could use recv instead of receive in some places here

**Review by jharveyb — APPROVED** [review]
Looks good, clean PR! 💯 

Needs another rebase.

File: tapgarden/custodian.go (hunk 1 above)
Comment (ffranr) [reply tapgarden/custodian.go]: > SendOrQuit

Good idea!

**Pull request merged**